



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for the Introduction to NGS workshop.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.6">
    
    
      
        <title>Quality Control - Introduction to NGS Workshop</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application.451f80e5.css">
      
      
    
    
      <script src="../../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#key-learning-outcomes" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../../.." title="Introduction to NGS Workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Introduction to NGS Workshop
              </span>
              <span class="md-header-nav__topic">
                Quality Control
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../../.." title="Introduction to NGS Workshop" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Introduction to NGS Workshop
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" checked>
    
    <label class="md-nav__link" for="nav-4">
      Hands-on Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Hands-on Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../cli/handout/handout/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Quality Control
      </label>
    
    <a href="./" title="Quality Control" class="md-nav__link md-nav__link--active">
      Quality Control
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" title="Author Information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-value-encoding-schema" title="Quality Value Encoding Schema" class="md-nav__link">
    Quality Value Encoding Schema
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#q-score-encoding-implemented-with-the-novaseq-platform" title="Q-score encoding implemented with the Novaseq platform" class="md-nav__link">
    Q-score encoding implemented with the Novaseq platform
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-visualisation" title="Quality Visualisation" class="md-nav__link">
    Quality Visualisation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#good-quality-data" title="Good Quality Data" class="md-nav__link">
    Good Quality Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#read-trimming" title="Read Trimming" class="md-nav__link">
    Read Trimming
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-based-trimming" title="Quality Based Trimming" class="md-nav__link">
    Quality Based Trimming
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-clipping" title="Adapter Clipping" class="md-nav__link">
    Adapter Clipping
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fixed-length-trimming" title="Fixed Length Trimming" class="md-nav__link">
    Fixed Length Trimming
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../alignment/handout/handout/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../rna-seq/handout/handout/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../chip-seq/handout/handout/" title="ChIP-Seq" class="md-nav__link">
      ChIP-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../de-novo/handout/handout/" title="De Novo" class="md-nav__link">
      De Novo
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../license/" title="License" class="md-nav__link">
      License
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#author-information" title="Author Information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-value-encoding-schema" title="Quality Value Encoding Schema" class="md-nav__link">
    Quality Value Encoding Schema
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#q-score-encoding-implemented-with-the-novaseq-platform" title="Q-score encoding implemented with the Novaseq platform" class="md-nav__link">
    Q-score encoding implemented with the Novaseq platform
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-visualisation" title="Quality Visualisation" class="md-nav__link">
    Quality Visualisation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#good-quality-data" title="Good Quality Data" class="md-nav__link">
    Good Quality Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#read-trimming" title="Read Trimming" class="md-nav__link">
    Read Trimming
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#quality-based-trimming" title="Quality Based Trimming" class="md-nav__link">
    Quality Based Trimming
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-clipping" title="Adapter Clipping" class="md-nav__link">
    Adapter Clipping
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fixed-length-trimming" title="Fixed Length Trimming" class="md-nav__link">
    Fixed Length Trimming
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Quality Control</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes</h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Assess the overall quality of NGS (FastQ format) sequence reads</p>
</li>
<li>
<p>Visualise the quality, and other associated matrices, of reads to
    decide on filters and cutoffs for cleaning up data ready for
    downstream analysis</p>
</li>
<li>
<p>Clean up adaptors and pre-process the sequence data for further
    analysis</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using</h2>
<h3 id="tools-used">Tools Used</h3>
<p>FastQC:<br />
<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></p>
<p>Skewer:<br />
<a href="http://sourceforge.net/projects/skewer/">http://sourceforge.net/projects/skewer/</a></p>
<p>FASTX-Toolkit:<br />
<a href="http://hannonlab.cshl.edu/fastx_toolkit/">http://hannonlab.cshl.edu/fastx_toolkit/</a></p>
<hr />
<h2 id="useful-links">Useful Links</h2>
<p>FASTQ Encoding: (<a href="http://en.wikipedia.org/wiki/FASTQ_format#Encoding">http://en.wikipedia.org/wiki/FASTQ_format#Encoding</a>)</p>
<hr />
<h2 id="author-information">Author Information</h2>
<p><em>Primary Author(s):</em>  <br />
Sonika Tyagi: <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#115;&#111;&#110;&#105;&#107;&#97;&#46;&#116;&#121;&#97;&#103;&#105;&#64;&#109;&#111;&#110;&#97;&#115;&#104;&#46;&#101;&#100;&#117;">&#115;&#111;&#110;&#105;&#107;&#97;&#46;&#116;&#121;&#97;&#103;&#105;&#64;&#109;&#111;&#110;&#97;&#115;&#104;&#46;&#101;&#100;&#117;</a></p>
<p><em>Contributor(s):</em>  <br />
Nandan Deshpande: <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#110;&#46;&#100;&#101;&#115;&#104;&#112;&#97;&#110;&#100;&#101;&#64;&#117;&#110;&#115;&#119;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#110;&#46;&#100;&#101;&#115;&#104;&#112;&#97;&#110;&#100;&#101;&#64;&#117;&#110;&#115;&#119;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a></p>
<hr />
<h2 id="introduction">Introduction</h2>
<p>Going on a blind date with your read set? For a better understanding of
the consequences please check the data quality!</p>
<p>For the purpose of this tutorial we are focusing only on Illumina
sequencing which uses ’sequence by synthesis’ technology in a highly
parallel fashion. Although Illumina high throughput sequencing provides
highly accurate sequence data, several sequence artifacts, including
base calling errors and small insertions/deletions, poor quality reads
and primer/adapter contamination are quite common in the high throughput
sequencing data. The primary errors are substitution errors. The error
rates can vary from 0.5-2.0% with errors mainly rising in frequency at
the 3’ ends of reads.</p>
<p>One way to investigate sequence data quality is to visualize the quality
scores and other metrics in a compact manner to get an idea about the
quality of a read data set. Read data sets can be improved by pre
processing in different ways like trimming off low quality bases,
cleaning up any sequencing adapters, removing PCR duplicates and
screening for contamination. We can also look at other statistics such
as, sequence length distribution, base composition, sequence complexity,
presence of ambiguous bases etc. to assess the overall quality of the
data set.</p>
<p>Highly redundant coverage (&gt;15X) of the genome can be used to correct
sequencing errors in the reads before assembly. Various k-mer based
error correction methods exist but are beyond the scope of this
tutorial.</p>
<h3 id="quality-value-encoding-schema">Quality Value Encoding Schema</h3>
<p>Quality scoring calculates a set of predictors for each base call, and then uses the predictor
values to look up the Q-score in a quality table. Quality tables are created to provide optimally
accurate quality predictions for runs generated by a specific configuration of sequencing
platform and version of chemistry (<a href="http://www.illumina.com">www.illumina.com</a>).
In order to use a single character to encode Phred qualities, <a href="http://shop.alterlinks.com/ascii-table/ascii-table-us.php">ASCII characters</a> are used. All ASCII characters have a decimal number associated with them but the first 32
characters are non-printable (e.g. backspace, shift, return, escape).
Therefore, the first printable ASCII character is number 33, the
exclamation mark (<code class="codehilite">!</code>). In Phred+33 encoded quality values the exclamation
mark takes the Phred quality score of zero.</p>
<p>Early Solexa (now Illumina) sequencing needed to encode negative quality
values. Because ASCII characters &lt; 33 are non-printable, using the
Phred+33 encoding was not possible. Therefore, they simply moved the
offset from 33 to 64 thus inventing the Phred+64 encoded quality values.
In this encoding a Phred quality of zero is denoted by the ASCII number
64 (the @ character). Since Illumina 1.8, quality values are now encoded
using Phred+33.</p>
<p>FASTQ does not provide a way to describe what quality encoding is used
for the quality values. Therefore, you should find this out from your
sequencing provider. Alternatively, you may be able to figure this out
by determining what ASCII characters are present in the FASTQ file. E.g
the presence of numbers in the quality strings, can only mean the
quality values are Phred+33 encoded. However, due to the overlapping
nature of the Phred+33 and Phred+64 encoding schema it is not always
possible to identify what encoding is in use. For example, if the only
characters seen in the quality string are (<code class="codehilite">@ABCDEFGHI</code>), then it is
impossible to know if you have really good Phred+33 encoded qualities or
really bad Phred+64 encoded qualities.</p>
<p>For a graphical representation of the different ASCII characters used in
the two encoding schema see:
(<a href="http://en.wikipedia.org/wiki/FASTQ_format#Encoding">http://en.wikipedia.org/wiki/FASTQ_format#Encoding</a>).</p>
<h3 id="q-score-encoding-implemented-with-the-novaseq-platform">Q-score encoding implemented with the Novaseq platform</h3>
<p>In order to reduce the data footprints Illumina has come up with a new
method to reduce quality score resolution and optimise data storae. The new Q-score
encoding now follows an 8 level mapping of individual quality scores (0-40 or &gt;40) [See <strong>Table 1</strong>].
With the new scoring scheme the original scores 20-24 may form one bin and the quality scores in that
bin mapped to a new value of 22. This can be thought of as simply replacing all the
occurrences of scores 20, 21, 23, 24 with a new score of 22 in the output sequence.
Illumina claims that with the new Q-scoring system the reduction in the Illumina raw sequence format (.bcl) is typically &gt; 50% and the resulting sorted BAM  les are reduced by ~30%.</p>
<table>
<thead>
<tr>
<th align="left">Quality Score Bins</th>
<th align="left">Mapped quality scores</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">N (no call)</td>
<td align="left">N (no call)</td>
</tr>
<tr>
<td align="left">2-9</td>
<td align="left">6</td>
</tr>
<tr>
<td align="left">10-19</td>
<td align="left">15</td>
</tr>
<tr>
<td align="left">20-24</td>
<td align="left">22</td>
</tr>
<tr>
<td align="left">25-29</td>
<td align="left">27</td>
</tr>
<tr>
<td align="left">30-34</td>
<td align="left">33</td>
</tr>
<tr>
<td align="left">35-39</td>
<td align="left">37</td>
</tr>
<tr>
<td align="left">>=40</td>
<td align="left">40</td>
</tr>
</tbody>
</table>
<p><strong>Table 1:</strong> Novaseq Q-score bins mapping</p>
<hr />
<h2 id="prepare-the-environment">Prepare the Environment</h2>
<p>To investigate sequence data quality we will demonstrate tools called
FastQC and Skewer. FastQC will process and present the reports in a
visual manner. Based on the results, the sequence data can be processed
using the Skewer. We will use one data set in this practical, which can
be found in the QC directory on your desktop.</p>
<p>Open the Terminal and go to the directory where the data are stored:</p>
<div class="codehilite"><pre><span></span>cd
ls
cd qc
pwd
</pre></div>


<p>At any time, help can be displayed for FastQC using the following
command:</p>
<div class="codehilite"><pre><span></span>fastqc -h
</pre></div>


<p>Look at SYNOPSIS (Usage) and options after typing <code class="codehilite">fastqc -h</code></p>
<hr />
<h2 id="quality-visualisation">Quality Visualisation</h2>
<p>We have a file for a good quality and bad quality statistics. FastQC
generates results in the form of a zipped and unzipped directory for
each input file.</p>
<p>Execute the following command on the two files:</p>
<div class="codehilite"><pre><span></span>fastqc -f fastq qcdemo_R1.fastq.gz
fastqc -f fastq qcdemo_R2.fastq.gz
</pre></div>


<p>View the FastQC report file of the bad data using a web browser such as
firefox. The <code class="codehilite">&amp;</code> sign puts the job in the background.</p>
<div class="codehilite"><pre><span></span>firefox qcdemo_R2_fastqc.html &amp;amp;
</pre></div>


<p><br>
The report file will have a Basic Statistics table and various graphs
and tables for different quality statistics e.g.:</p>
<table>
<thead>
<tr>
<th align="left">Property</th>
<th align="left">Value</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">Filename</td>
<td align="left">qcdemo_R2.fastq.gz</td>
</tr>
<tr>
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr>
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr>
<td align="left">Total Sequences</td>
<td align="left">1000000</td>
</tr>
<tr>
<td align="left">Filtered Sequences</td>
<td align="left">0</td>
</tr>
<tr>
<td align="left">Sequence length</td>
<td align="left">150</td>
</tr>
<tr>
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><strong>Table 2:</strong> Summary statistics for bad_example_untrimmed</p>
<p><br>
<img alt="image" src="../images/bad_qcdemo_R2.png" /></p>
<p><strong>Figure 1:</strong> bad_example_untrimmed_QC_plot</p>
<p>A Phred quality score (or Q-score) expresses an error probability. In
particular, it serves as a convenient and compact way to communicate
very small error probabilities. The probability that base A is wrong
(P(A)) is expressed by a quality score, Q(A), according to the
relationship:</p>
<p>
<script type="math/tex; mode=display">Q(A) =-10 log10(P(A))</script>
</p>
<p>The relationship between the quality score and error probability is
demonstrated with the following table:</p>
<table>
<thead>
<tr>
<th>Quality score, Q(A)</th>
<th align="left">Error probability, P(A)</th>
<th align="left">Accuracy of base call</th>
</tr>
</thead>
<tbody>
<tr>
<td>10</td>
<td align="left">0.1</td>
<td align="left">90%</td>
</tr>
<tr>
<td>20</td>
<td align="left">0.01</td>
<td align="left">99%</td>
</tr>
<tr>
<td>30</td>
<td align="left">0.001</td>
<td align="left">99.9%</td>
</tr>
<tr>
<td>40</td>
<td align="left">0.0001</td>
<td align="left">99.99%</td>
</tr>
<tr>
<td>50</td>
<td align="left">0.00001</td>
<td align="left">99.999%</td>
</tr>
</tbody>
</table>
<p><strong>Table 3:</strong> Quality Error Probabilities</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How many sequences were there in your file? What is the read length?</p>
<div class="admonition success">
<details><summary>Answer</summary><p>1,000,000. read length=150bp</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Does the quality score values vary throughout the read length?</p>
<div class="admonition hint">
<details><summary>Hint</summary><p>Look at the ’per base sequence quality plot’</p>
</details>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Yes. Quality scores are dropping towards the end of the reads.</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What is the quality score range you see?</p>
<div class="admonition success">
<details><summary>Answer</summary><p>2-40</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>At around which position do the scores start falling below Q20 for the 25% quartile range (25%of reads below Q20)?</p>
<div class="admonition success">
<details><summary>Answer</summary><p>Around 30 bp position</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How can we trim the reads to filter out the low quality data?</p>
<div class="admonition success">
<details><summary>Answer</summary><p>By trimming off the bases after a fixed position of the read or by trimming off bases based on the quality score.</p>
</details>
</div>
</div>
<h3 id="good-quality-data">Good Quality Data</h3>
<p>View the FastQC report files <code class="codehilite">fastqc_report.html</code> to see examples of a
good quality data and compare the quality plot with that of the
<code class="codehilite">bad_example_fastqc</code>.</p>
<div class="codehilite"><pre><span></span>firefox qcdemo_R1_fastqc.html &amp;amp;
</pre></div>


<p>Sequencing errors can complicate the downstream analysis, which normally
requires that reads be aligned to each other (for genome assembly) or to
a reference genome (for detection of mutations). Sequence reads
containing errors may lead to ambiguous paths in the assembly or
improper gaps. In variant analysis projects sequence reads are aligned
against the reference genome. The errors in the reads may lead to more
mismatches than expected from mutations alone. But if these errors can
be removed or corrected, the read alignments and hence the variant
detection will improve. The assemblies will also improve after
pre-processing the reads to remove errors.</p>
<hr />
<h2 id="read-trimming">Read Trimming</h2>
<p>Read trimming can be done in a variety of different ways. Choose a
method which best suits your data. Here we are giving examples of
fixed-length trimming and quality-based trimming.</p>
<h3 id="quality-based-trimming">Quality Based Trimming</h3>
<p>Base call quality scores can be used to dynamically determine the trim
points for each read. A quality score threshold and minimum read length
following trimming can be used to remove low quality data.</p>
<p>The previous FastQC results show R1 is fine but R2 has low quality at
the end. There is no adaptor contamination though. We will be using
Skewer to perform the quality trimming.</p>
<p>Run the following command to quality trim a set of paired end data.</p>
<div class="codehilite"><pre><span></span>cd /home/trainee/qc
skewer -t 4 -l 50  -q 30 -Q 25 -m pe -o qcdemo qcdemo_R1.fastq.gz qcdemo_R2.fastq.gz
</pre></div>


<blockquote>
<p><strong>-t:</strong> number of threads to use<br />
<strong>-l:</strong> min length to keep after trimming<br />
<strong>-q:</strong> Quality threshold used for trimming at 3’ end<br />
<strong>-Q:</strong> mean quality threshold for a read<br />
<strong>-m:</strong> pair-end mode  </p>
</blockquote>
<p><br>
Run FastQC on the quality trimmed file and visualise the quality scores.</p>
<p>Look at the last files generated, are the file names same as the input ?</p>
<div class="codehilite"><pre><span></span>ls -ltr
</pre></div>


<p>Run Fastqc on the quality trimmed files:</p>
<div class="codehilite"><pre><span></span>fastqc -f fastq qcdemo-trimmed-pair1.fastq
fastqc -f fastq qcdemo-trimmed-pair2.fastq
</pre></div>


<p>Visualise the <code class="codehilite">fastqc</code> results:</p>
<div class="codehilite"><pre><span></span>firefox qcdemo-trimmed-pair1_fastqc.html &amp;amp;
firefox qcdemo-trimmed-pair2_fastqc.html &amp;amp;
</pre></div>


<p><br>
Let’s look at the quality from the second reads. The output should look
like:</p>
<table>
<thead>
<tr>
<th align="left">Property</th>
<th align="left">Value</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">Filename</td>
<td align="left">qcdemo-trimmed-pair2.fastq</td>
</tr>
<tr>
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr>
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr>
<td align="left">Total Sequences</td>
<td align="left">742262</td>
</tr>
<tr>
<td align="left">Filtered Sequences</td>
<td align="left">0</td>
</tr>
<tr>
<td align="left">Sequence length</td>
<td align="left">50-150</td>
</tr>
<tr>
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><strong>Table 4:</strong> Summary Statistics of QC_demo_R1_trimmed</p>
<p><br>
<img alt="image" src="../images/bad_qcdemo_R2_quality_trimmed.png" /></p>
<p><strong>Figure 2:</strong> bad_example_quality_trimmed_plot</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Did the number of total reads in R1 and R2 change after trimming?</p>
<div class="admonition success">
<details><summary>Answer</summary><p>Quality trimming discarded &gt;25000 reads. However, we retain a lot of
maximal length reads which have good quality all the way to the ends.</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What reads lengths were obtained after quality based trimming?</p>
<div class="admonition success">
<details><summary>Answer</summary><p>50-150<br />
Reads &lt;50 bp, following quality trimming, were discarded.</p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Did you observe adapter sequences in the data?</p>
<div class="admonition success">
<details><summary>Answer</summary><p>No. (Hint: look at the overrepresented sequences)    </p>
</details>
</div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How can you use -a option with fastqc? (Hint: try fastqc -h).</p>
<div class="admonition success">
<details><summary>Answer</summary><p>Adaptors can be supplied in a file for screening.</p>
</details>
</div>
</div>
<h3 id="adapter-clipping">Adapter Clipping</h3>
<p>Sometimes sequence reads may end up getting the leftover of adapters and
primers used in the sequencing process. It’s good practice to screen
your data for these possible contamination for more sensitive alignment
and assembly based analysis.</p>
<p>This is particularly important when read lengths can be longer than the
molecules being sequenced. For example when sequencing miRNAs.</p>
<p>Various QC tools are available to screen and/or clip these
adapter/primer sequences from your data. Apart from <code class="codehilite">skewer</code> which will be
using today the following two tools are also useful for trimming and
removing adapter sequence:</p>
<ul>
<li>Cutadapt: <a href="http://code.google.com/p/cutadapt/">http://code.google.com/p/cutadapt/</a></li>
<li>Trimmomatic: <a href="http://www.usadellab.org/cms/?page=trimmomatic">http://www.usadellab.org/cms/?page=trimmomatic</a></li>
</ul>
<p>Here we are demonstrating <code class="codehilite">Skewer</code> to trim a given adapter sequence.</p>
<div class="codehilite"><pre><span></span>cd /home/trainee/qc
fastqc -f fastq  adaptorQC.fastq.gz
firefox adaptorQC_fastqc.html
skewer -x TGGAATTCTCGGGTGCCAAGGT -t 20 -l 10 -L 35 -q 30 adaptorQC.fastq.gz
</pre></div>


<blockquote>
<p><strong>-x:</strong> adaptor sequence used<br />
<strong>-t:</strong> number of threads to use<br />
<strong>-l:</strong> min length to keep after trimming<br />
<strong>-L:</strong> Max length to keep after trimming, in this experiment we were expecting only small RNA fragments<br />
<strong>-Q:</strong> Quality threshold used for trimming at 3’ end. Use -m option to control the end you want to trim  </p>
</blockquote>
<p><br>
Run FastQC on the adapter trimmed file and visualise the quality scores.
Fastqc now shows adaptor free results.</p>
<div class="codehilite"><pre><span></span>fastqc adaptorQC.fastq-trimmed.fastq
firefox adaptorQC.fastq-trimmed_fastqc.html &amp;amp;
</pre></div>


<h3 id="fixed-length-trimming">Fixed Length Trimming</h3>
<p><strong>We will not cover fixed length trimming but provide the following for
your information.</strong> Low quality read ends can be trimmed using a
fixed-length trimming. We will use the <code class="codehilite">fastx_trimmer</code> from the
FASTX-Toolkit. Usage message to find out various options you can use
with this tool. Type <code class="codehilite">fastx_trimmer -h</code> at anytime to display help.</p>
<p>We will now do fixed-length trimming of the <code class="codehilite">bad_example.fastq</code> file
using the following command. You should still be in the qc directory, if
not cd back in.</p>
<div class="codehilite"><pre><span></span>cd /home/trainee/qc
fastqc -f fastq bad_example.fastq
fastx_trimmer -h
fastx_trimmer -Q 33 -f 1 -l 80 -i bad_example.fastq -o bad_example_trimmed01.fastq
</pre></div>


<p>We used the following options in the command above:</p>
<blockquote>
<p><strong>-Q 33:</strong> Indicates the input quality scores are Phred+33 encoded<br />
<strong>-f:</strong> First base to be retained in the output<br />
<strong>-l:</strong> Last base to be retained in the output<br />
<strong>-i:</strong> Input FASTQ file name<br />
<strong>-o:</strong> Output file name  </p>
</blockquote>
<p><br>
Run FastQC on the trimmed file and visualise the quality scores of the
trimmed file.</p>
<div class="codehilite"><pre><span></span>fastqc -f fastq bad_example_trimmed01.fastq
firefox bad_example_trimmed01_fastqc.html &amp;amp;
</pre></div>


<p>The output should look like:</p>
<table>
<thead>
<tr>
<th align="left">Property</th>
<th align="left">Value</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">Filename</td>
<td align="left">bad_example_trimmed01.fastq</td>
</tr>
<tr>
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr>
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr>
<td align="left">Total Sequences</td>
<td align="left">40000</td>
</tr>
<tr>
<td align="left">Filtered Sequences</td>
<td align="left">0</td>
</tr>
<tr>
<td align="left">Sequence length</td>
<td align="left">80</td>
</tr>
<tr>
<td align="left">%GC</td>
<td align="left">48</td>
</tr>
</tbody>
</table>
<p><strong>Table 5:</strong> Summary Statistics of bad_example_trimmed summary</p>
<p><br>
<img alt="image" src="../images/bad_example_trimmed_to_80bp.png" title="reaaaaaa" /></p>
<p><strong>Figure 3:</strong> bad_example_trimmed_plot</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What values would you use for <code class="codehilite">-f</code> if you wanted to trim off 10 bases at the 5’ end of the reads?</p>
<div class="admonition success">
<details><summary>Answer</summary><p><code class="codehilite">-f 11</code></p>
</details>
</div>
</div>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../../cli/handout/handout/" title="Introduction to Command Line" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Introduction to Command Line
              </span>
            </div>
          </a>
        
        
          <a href="../../../alignment/handout/handout/" title="Read Alignment" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Read Alignment
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application.5e60981f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>