



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for the Introduction to NGS workshop.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-3.0.6">
    
    
      
        <title>De Novo Genome Assembly - Introduction to NGS Workshop</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application.451f80e5.css">
      
      
    
    
      <script src="../../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#key-learning-outcomes" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../../.." title="Introduction to NGS Workshop" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Introduction to NGS Workshop
              </span>
              <span class="md-header-nav__topic">
                De Novo Genome Assembly
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../../.." title="Introduction to NGS Workshop" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Introduction to NGS Workshop
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" checked>
    
    <label class="md-nav__link" for="nav-4">
      Hands-on Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Hands-on Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../cli/handout/handout/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../qc/handout/handout/" title="Quality Control" class="md-nav__link">
      Quality Control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../alignment/handout/handout/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../rna-seq/handout/handout/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        De Novo Genome Assembly
      </label>
    
    <a href="./" title="De Novo Genome Assembly" class="md-nav__link md-nav__link--active">
      De Novo Genome Assembly
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#downloading-and-compiling-velvet" title="Downloading and Compiling Velvet" class="md-nav__link">
    Downloading and Compiling Velvet
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembling-paired-end-reads-using-velvet" title="Assembling Paired-end Reads using Velvet" class="md-nav__link">
    Assembling Paired-end Reads using Velvet
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#amos-hawkeye" title="AMOS Hawkeye" class="md-nav__link">
    AMOS Hawkeye
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#velvet-and-data-quality" title="Velvet and Data Quality" class="md-nav__link">
    Velvet and Data Quality
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../denovo-canu/handout/handout/" title="De Novo Canu" class="md-nav__link">
      De Novo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../eukaryotic/handout/handout/" title="Eukaryote Genome Assembly" class="md-nav__link">
      Eukaryote Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../scaffolding/handout/handout/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../license/" title="License" class="md-nav__link">
      License
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#downloading-and-compiling-velvet" title="Downloading and Compiling Velvet" class="md-nav__link">
    Downloading and Compiling Velvet
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembling-paired-end-reads-using-velvet" title="Assembling Paired-end Reads using Velvet" class="md-nav__link">
    Assembling Paired-end Reads using Velvet
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#amos-hawkeye" title="AMOS Hawkeye" class="md-nav__link">
    AMOS Hawkeye
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#velvet-and-data-quality" title="Velvet and Data Quality" class="md-nav__link">
    Velvet and Data Quality
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>De Novo Genome Assembly</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes</h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Compile velvet with appropriate compile-time parameters set for a
    specific analysis</p>
</li>
<li>
<p>Be able to choose appropriate assembly parameters</p>
</li>
<li>
<p>Assemble a set of paired-end reads from a single insert-size library</p>
</li>
<li>
<p>Be able to visualise an assembly in AMOS Hawkeye</p>
</li>
</ul>
<h2 id="resources-youll-be-using">Resources You’ll be Using</h2>
<p>Although we have provided you with an environment which contains all the
tools and data you will be using in this module, you may like to know
where we have sourced those tools and data from.</p>
<h3 id="tools-used">Tools Used</h3>
<p>Velvet:<br />
<a href="http://www.ebi.ac.uk/~zerbino/velvet/">http://www.ebi.ac.uk/~zerbino/velvet/</a></p>
<p>AMOS Hawkeye:<br />
<a href="http://apps.sourceforge.net/mediawiki/amos/index.php?title=Hawkeye">http://apps.sourceforge.net/mediawiki/amos/index.php?title=Hawkeye</a></p>
<p>gnx-tools:<br />
<a href="https://github.com/mh11/gnx-tools">https://github.com/mh11/gnx-tools</a></p>
<p>FastQC:<br />
<a href="http://www.bioinformatics.bbsrc.ac.uk/projects/fastqc/">http://www.bioinformatics.bbsrc.ac.uk/projects/fastqc/</a></p>
<p>R:<br />
<a href="http://www.r-project.org/">http://www.r-project.org/</a></p>
<h3 id="sources-of-data">Sources of Data</h3>
<ul>
<li>
<p><a href="ftp://ftp.ensemblgenomes.org/pub/release-8/bacteria/fasta/Staphylococcus/s_aureus_mrsa252/dna/s_aureus_mrsa252.EB1_s_aureus_mrsa252.dna.chromosome.Chromosome.fa.gz">ftp://ftp.ensemblgenomes.org/pub/release-8/bacteria/fasta/Staphylococcus/s_aureus_mrsa252/dna/s_aureus_mrsa252.EB1_s_aureus_mrsa252.dna.chromosome.Chromosome.fa.gz</a></p>
</li>
<li>
<p><a href="http://www.ebi.ac.uk/ena/data/view/SRX008042">http://www.ebi.ac.uk/ena/data/view/SRX008042</a></p>
</li>
<li>
<p><a href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR022/SRR022852/SRR022852_1.fastq.gz">ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR022/SRR022852/SRR022852_1.fastq.gz</a></p>
</li>
<li>
<p><a href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR022/SRR022852/SRR022852_2.fastq.gz">ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR022/SRR022852/SRR022852_2.fastq.gz</a></p>
</li>
<li>
<p><a href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR023/SRR023408/SRR023408_1.fastq.gz">ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR023/SRR023408/SRR023408_1.fastq.gz</a></p>
</li>
<li>
<p><a href="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR023/SRR023408/SRR023408_2.fastq.gz">ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR023/SRR023408/SRR023408_2.fastq.gz</a></p>
</li>
</ul>
<h2 id="introduction">Introduction</h2>
<p>The aim of this module is to become familiar with performing <em>de novo</em>
genome assembly using Velvet, a de Bruijn graph based assembler, on a
variety of sequence data.</p>
<h2 id="prepare-the-environment">Prepare the Environment</h2>
<p>The first exercise should get you a little more comfortable with the
computer environment and the command line.</p>
<p>First make sure that you are in the denovo working directory by typing:</p>
<div class="codehilite"><pre><span></span>cd /home/trainee/denovo
</pre></div>


<p>and making absolutely sure you’re there by typing:</p>
<div class="codehilite"><pre><span></span>pwd
</pre></div>


<p>Now create sub-directories for this and the two other velvet practicals.
All these directories will be made as sub-directories of a directory for
the whole course called NGS. For this you can use the following
commands:</p>
<div class="codehilite"><pre><span></span>mkdir -p NGS/velvet/{part1,part2}
</pre></div>


<p>The <code class="codehilite">-p</code> tells <code class="codehilite">mkdir</code> (make directory) to make any parent directories
if they don’t already exist. You could have created the above
directories one-at-a-time by doing this instead:</p>
<div class="codehilite"><pre><span></span>mkdir NGS
mkdir NGS/velvet
mkdir NGS/velvet/part1
mkdir NGS/velvet/part2
</pre></div>


<p>After creating the directories, examine the structure and move into the
directory ready for the first velvet exercise by typing:</p>
<div class="codehilite"><pre><span></span>ls -R NGS
cd NGS/velvet/part1
pwd
</pre></div>


<h2 id="downloading-and-compiling-velvet">Downloading and Compiling Velvet</h2>
<p>For the duration of this workshop, all the software you require has been
set up for you already. This might not be the case when you return to
“real life”. Many of the programs you will need, including velvet, are
quite easy to set up, it might be instructive to try a couple.</p>
<p>Although you will be using the preinstalled version of velvet, it is
useful to know how to compile velvet as some of the parameters you might
like to control can only be set at compile time. You can find the latest
version of velvet at:</p>
<p><a href="http://www.ebi.ac.uk/~zerbino/velvet/">http://www.ebi.ac.uk/~zerbino/velvet/</a></p>
<p>You could go to this URL and download the latest velvet version, or
equivalently, you could type the following, which will download, unpack,
inspect, compile and execute your locally compiled version of velvet:</p>
<div class="codehilite"><pre><span></span>cd /home/trainee/denovo/NGS/velvet/part1
pwd
tar xzf /home/trainee/denovo/data/velvet_1.2.10.tgz
ls -R
cd velvet_1.2.10
make
./velveth
</pre></div>


<p>The standout displayed to screen when ’make’ runs may contain an error
message but it is ignored</p>
<p>Take a look at the executables you have created. They will be displayed
as green by the command:</p>
<div class="codehilite"><pre><span></span>ls --color=always
</pre></div>


<p>The switch <code class="codehilite">–color</code>, instructs that files be coloured according to their
type. This is often the default but we are just being explicit. By
specifying the value <code class="codehilite">always</code>, we ensure that colouring is always
applied, even from a script.</p>
<p>Have a look of the output the command produces and you will see that
<code class="codehilite">MAXKMERLENGTH=31</code> and <code class="codehilite">CATEGORIES=2</code> parameters were passed into the
compiler.</p>
<p>This indicates that the default compilation was set for de Bruijn graph
k-mers of maximum size 31 and to allow a maximum of just 2 read
categories. You can override these, and other, default configuration
choices using command line parameters. Assume, you want to run velvet
with a k-mer length of 41 using 3 categories, velvet needs to be
recompiled to enable this functionality by typing:</p>
<div class="codehilite"><pre><span></span>make clean
make MAXKMERLENGTH=41 CATEGORIES=3
./velveth
</pre></div>


<p>Discuss with the persons next to you the following questions:</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What are the consequences of the parameters you have given make for velvet?</p>
</div>
<div class="admonition success">
<details><summary><strong>Answer</strong></summary><p>MAXKMERLENGTH: increase the max k-mer length from 31 to 41   <br />
CATEGORIES: paired-end data require to be put into separate categories. By increasing this parameter from 2 to 3 allows you to process 3 paired / mate-pair libraries and unpaired data.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why does Velvet use k-mer 31 and 2 categories as default?</p>
</div>
<div class="admonition success">
<details><summary><strong>Answer</strong></summary><p>Possibly a number of reason:    <br />
1) odd number to avoid palindromes      <br />
2) The first reads were very short (20-40 bp) and there were hardly any paired-end data around so there was no need to allow for longer k-mer lengths / more categories.      <br />
3) For programmers: 31 bp get stored in 64 bits (using 2bit encoding)</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Should you get better results by using a longer k-mer length?</p>
</div>
<div class="admonition success">
<details><summary><strong>Answer</strong></summary><p>If you can achieve a good k-mer coverage - yes.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What effect would the following compile-time parameters have on velvet:
OPENMP=Y</p>
</div>
<div class="admonition success">
<details><summary><strong>Answer</strong></summary><p>Turn on multithreading</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>LONGSEQUENCES=Y</p>
</div>
<div class="admonition success">
<details><summary><strong>Answer</strong></summary><p>Assembling reads / contigs longer than 32kb long</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>BIGASSEMBLY=Y</p>
</div>
<div class="admonition success">
<details><summary><strong>Answer</strong></summary><p>Using more than 2.2 billion reads</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>SINGLE_COV_CAT=Y</p>
</div>
<div class="admonition success">
<details><summary><strong>Answer</strong></summary><p>Merge all coverage statistics into a single variable - save memory</p>
</details>
</div>
<p>For a further description of velvet compile and runtime parameters
please see the velvet Manual:
<a href="https://github.com/dzerbino/velvet/wiki/Manual">https://github.com/dzerbino/velvet/wiki/Manual</a></p>
<h2 id="assembling-paired-end-reads-using-velvet">Assembling Paired-end Reads using Velvet</h2>
<p>The use of paired-end data in <em>de novo</em> genome assembly results in
better quality assemblies, particularly for larger, more complex
genomes. In addition, paired-end constraint violation (expected distance
and orientation of paired reads) can be used to identify misassemblies.</p>
<p>If you are doing <em>de novo</em> assembly, pay the extra and get paired-ends:
they’re worth it!</p>
<p>The data you will examine in this exercise is again from Staphylococcus
aureus which has a genome of around 3MBases. The reads are Illumina
paired end with an insert size of ~350 bp.</p>
<p>The required data can be downloaded from the SRA. Specifically, the run
data (SRR022852) from the SRA Sample SRS004748.</p>
<p>The following exercise focuses on preparing the paired-end FASTQ files
ready for Velvet, using Velvet in paired-end mode and comparing results
with Velvet’s ’auto’ option.</p>
<p>First move to the directory you made for this exercise and make a
suitable named directory for the exercise:</p>
<div class="codehilite"><pre><span></span>cd /home/trainee/denovo/NGS/velvet/part2 
mkdir SRS004748 
cd SRS004748
</pre></div>


<p>There is no need to download the read files, as they are already stored
locally. You will simply create a symlink to this pre-downloaded data
using the following commands:</p>
<div class="codehilite"><pre><span></span>ln -s /home/trainee/denovo/data/SRR022852_?.fastq.gz ./
</pre></div>


<p>It is interesting to monitor the computer’s resource utilisation,
particularly memory. A simple way to do this is to open a second
terminal and in it type:</p>
<div class="codehilite"><pre><span></span>top
</pre></div>


<p><code class="codehilite">top</code> is a program that continually monitors all the processes running
on your computer, showing the resources used by each. Leave this running
and refer to it periodically throughout your Velvet analyses.
Particularly if they are taking a long time or whenever your curiosity
gets the better of you. You should find that as this practical
progresses, memory usage will increase significantly.</p>
<p>Now, back to the first terminal, you are ready to run <code class="codehilite">velveth</code> and
<code class="codehilite">velvetg</code>. The reads are <code class="codehilite">-shortPaired</code> and for the first run you should
not use any parameters for <code class="codehilite">velvetg</code>.</p>
<p>From this point on, where it will be informative to time your runs. This
is very easy to do, just prefix the command to run the program with the
command <code class="codehilite">time</code>. This will cause UNIX to report how long the program took
to complete its task.</p>
<p>Set the two stages of velvet running, whilst you watch the memory usage
as reported by <code class="codehilite">top</code>. Time the <code class="codehilite">velvetg</code> stage:</p>
<div class="codehilite"><pre><span></span>velveth run_25 25 -fmtAuto -create_binary -shortPaired -separate SRR022852_1.fastq.gz SRR022852_2.fastq.gz
time velvetg run_25
</pre></div>


<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What does <code class="codehilite">-fmtAuto</code> and <code class="codehilite">-create_binary</code> do? (see help menu)</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p><code class="codehilite">-fmtAuto</code> tries to detect the correct format of the input files e.g. FASTA, FASTQ and whether they are compressed or not.   <br />
<code class="codehilite">-create_binary</code> outputs sequences as a binary file. That means that <code class="codehilite">velvetg</code> can read the sequences from the binary file more quickly that from the original sequence files.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Comment on the use of memory and CPU for <code class="codehilite">velveth</code> and <code class="codehilite">velvetg</code>?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p><code class="codehilite">velveth</code> uses only one CPU while <code class="codehilite">velvetg</code> uses all possible CPUs for some parts of the calculation.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How long did <code class="codehilite">velvetg</code> take?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>My own measurements are:   <br />
<code class="codehilite">real 1m8.877s; user 4m15.324s; sys 0m4.716s</code></p>
</details>
</div>
<p>Next, after saving your <code class="codehilite">contigs.fa</code> file from being overwritten, set
the cut-off parameters that you investigated in the previous exercise
and rerun <code class="codehilite">velvetg</code>. time and monitor the use of resources as
previously. Start with <code class="codehilite">-cov_cutoff 16</code> thus:</p>
<div class="codehilite"><pre><span></span>mv run_25/contigs.fa run_25/contigs.fa.0
time velvetg run_25 -cov_cutoff 16
</pre></div>


<p>Up until now, <code class="codehilite">velvetg</code> has ignored the paired-end information. Now try
running <code class="codehilite">velvetg</code> with both <code class="codehilite">-cov_cutoff 16</code> and <code class="codehilite">-exp_cov 26</code>, but
first save your <code class="codehilite">contigs.fa</code> file. By using <code class="codehilite">-cov_cutoff</code> and
<code class="codehilite">-exp_cov</code>, <code class="codehilite">velvetg</code> tries to estimate the insert length, which you
will see in the <code class="codehilite">velvetg</code> output. The command is, of course:</p>
<div class="codehilite"><pre><span></span>mv run_25/contigs.fa run_25/contigs.fa.1
time velvetg run_25 -cov_cutoff 16 -exp_cov 26
</pre></div>


<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Comment on the time required, use of memory and CPU for <code class="codehilite">velvetg</code>?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Runtime is lower when velvet can reuse previously calculated data. By using <code class="codehilite">-exp_cov</code>, the memory usage increases.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Which insert length does Velvet estimate?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Paired-end library 1 has length: 228, sample standard deviation: 26</p>
</details>
</div>
<p>Next try running <code class="codehilite">velvetg</code> in ‘paired-end mode‘. This entails running
<code class="codehilite">velvetg</code> specifying the insert length with the parameter <code class="codehilite">-ins_length</code>
set to 350. Even though velvet estimates the insert length it is always
advisable to check / provide the insert length manually as velvet can
get the statistics wrong due to noise. Just in case, save your last
version of <code class="codehilite">contigs.fa</code>. The commands are:</p>
<div class="codehilite"><pre><span></span>mv run_25/contigs.fa run_25/contigs.fa.2
time velvetg run_25 -cov_cutoff 16 -exp_cov 26 -ins_length 350
mv run_25/contigs.fa run_25/contigs.fa.3
</pre></div>


<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How fast was this run?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>My own measurements are:   <br />
<code class="codehilite">real 0m29.792s; user 1m4.372s; sys 0m3.880s</code></p>
</details>
</div>
<p>Take a look into the Log file.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What is the N50 value for the <code class="codehilite">velvetg</code> runs using the switches</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Base run: 19,510 bp  <br />
<code class="codehilite">-cov_cutoff 16</code>: 24,739 bp   <br />
<code class="codehilite">-cov_cutoff 16 -exp_cov 26</code>: 61,793 bp   <br />
<code class="codehilite">-cov_cutoff 16 -exp_cov 26 -ins_length 350</code>: n50 of 62,740 bp; max 194,649 bp; total 2,871,093 bp     </p>
</details>
</div>
<p>Try giving the <code class="codehilite">-cov_cutoff</code> and/or <code class="codehilite">-exp_cov</code> parameters the value
<code class="codehilite">auto</code>. See the <code class="codehilite">velvetg</code> help to show you how. The information Velvet
prints during running includes information about the values used
(coverage cut-off or insert length) when using the <code class="codehilite">auto</code> option.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What coverage values does Velvet choose (hint: look at the output that Velvet produces while running)?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Median coverage depth = 26.021837  <br />
Removing contigs with coverage &lt; 13.010918 …</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How does the N50 value change?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>n50 of 68,843 bp; max 194,645 bp; total 2,872,678 bp</p>
</details>
</div>
<p>Run <code class="codehilite">gnx</code> on all the <code class="codehilite">contig.fa</code> files you have generated in the course
of this exercise. The command will be:</p>
<div class="codehilite"><pre><span></span>gnx -min 100 -nx 25,50,75 run_25/contigs.fa*
</pre></div>


<div class="admonition note">
<p class="admonition-title">Question</p>
<p>For which runs are there Ns in the <code class="codehilite">contigs.fa</code> file and why?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>contigs.fa.2, contigs.fa.3, contigs.fa</p>
</details>
</div>
<p>Velvet tries to use the provided (or infers) the insert length and fills
ambiguous regions with Ns.</p>
<p>Comment on the number of contigs and total length generated for each
run.</p>
<table>
<thead>
<tr>
<th>Filename</th>
<th>No. contigs</th>
<th align="right">Total length</th>
<th align="center">No. Ns</th>
</tr>
</thead>
<tbody>
<tr>
<td>Contigs.fa.0</td>
<td>631</td>
<td align="right">2,830,659</td>
<td align="center">0</td>
</tr>
<tr>
<td>Contigs.fa.1</td>
<td>580</td>
<td align="right">2,832,670</td>
<td align="center">0</td>
</tr>
<tr>
<td>Contigs.fa.2</td>
<td>166</td>
<td align="right">2,849,919</td>
<td align="center">4,847</td>
</tr>
<tr>
<td>Contigs.fa.3</td>
<td>166</td>
<td align="right">2,856,795</td>
<td align="center">11,713</td>
</tr>
<tr>
<td>Contigs.fa</td>
<td>163</td>
<td align="right">2,857,439</td>
<td align="center">11,526</td>
</tr>
</tbody>
</table>
<h2 id="amos-hawkeye">AMOS Hawkeye</h2>
<p>We will now output the assembly in the AMOS massage format and visualise
the assembly using AMOS Hawkeye.</p>
<p>Run <code class="codehilite">velvetg</code> with appropriate arguments and output the AMOS message
file, then convert it to an AMOS bank and open it in Hawkeye:</p>
<div class="codehilite"><pre><span></span>time velvetg run_25 -cov_cutoff 16 -exp_cov 26 -ins_length 350 -amos_file yes -read_trkg yes 
time bank-transact -c -b run_25/velvet_asm.bnk -m run_25/velvet_asm.afg         
hawkeye run_25/velvet_asm.bnk
</pre></div>


<p>Looking at the scaffold view of a contig, comment on the proportion of
“happy mates” to “compressed mates” and “stretched mates”.</p>
<p>Nearly all mates are compressed with no stretched mates and very few
happy mates.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What is the mean and standard deviation of the insert size reported under the Libraries tab?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Mean: 350 bp SD: 35 bp</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Look at the actual distribution of insert sizes for this library. Can you explain where there is a difference between the mean and SD reported in those two places?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>We specified <code class="codehilite">-ins_length 350</code> to the <code class="codehilite">velvetg</code> command. Velvet uses this value, in the AMOS message file that it outputs, rather than its own estimate.</p>
</details>
</div>
<p>You can get AMOS to re-estimate the mean and SD of insert sizes using
intra-contig pairs. First, close Hawkeye and then run the following
commands before reopening the AMOS bank to see what has changed.</p>
<div class="codehilite"><pre><span></span>asmQC -b run_25/velvet_asm.bnk -scaff -recompute -update -numsd 2
hawkeye run_25/velvet_asm.bnk
</pre></div>


<p>Looking at the scaffold view of a contig, comment on the proportion of
“happy mates” to “compressed mates” and “stretched mates”.</p>
<p>There are only a few compressed and stretched mates compared to happy
mates. There are similar numbers of stretched and compressed mates.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What is the mean and standard deviation of the insert size reported under the Libraries tab?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Mean: 226 bp SD: 25 bp</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Look at the actual distribution of insert sizes for this library. Does the mean and SD reported in both places now match?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Yes</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Can you find a region with an unusually high proportion of stretched, compressed, incorrectly orientated or linking mates? What might this situation indicate?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>This would indicate a possible misassembly and worthy of further investigation.   <br />
Look at the largest scaffold, there are stacks of stretched pairs which span contig boundaries. This indicates that the gap size has been underestimated during the scaffolding phase.</p>
</details>
</div>
<h2 id="velvet-and-data-quality">Velvet and Data Quality</h2>
<p>So far we have used the raw read data without performing any quality
control or read trimming prior to doing our velvet assemblies.</p>
<p>Velvet does not use quality information present in FASTQ files.</p>
<p>For this reason, it is vitally important to perform read QC and quality
trimming. In doing so, we remove errors/noise from the dataset which in
turn means velvet will run faster, will use less memory and will produce
a better assembly. Assuming we haven’t compromised too much on coverage.</p>
<p>To investigate the effect of data quality, we will use the run data
(SRR023408) from the SRA experiment SRX008042. The reads are Illumina
paired end with an insert size of 92 bp.</p>
<p>Go back to the main directory for this exercise and create and enter a
new directory dedicated to this phase of the exercise. The commands are:</p>
<div class="codehilite"><pre><span></span>cd /home/trainee/denovo/NGS/velvet/part2 
mkdir SRX008042 
cd SRX008042
</pre></div>


<p>Create symlinks to the read data files that we downloaded for you from
the SRA:</p>
<div class="codehilite"><pre><span></span>ln -s /home/trainee/denovo/data/SRR023408_?.fastq.gz ./
</pre></div>


<p>We will use FastQC, a tool you should be familiar with, to visualise the
quality of our data. We will use FastQC in the Graphical User Interface
(GUI) mode.</p>
<p>Start FastQC and set the process running in the background, by using a
trailing <code class="codehilite">&amp;</code>, so we get control of our terminal back for entering more
commands:</p>
<div class="codehilite"><pre><span></span>fastqc &amp;amp;
</pre></div>


<p>Open the two compressed FASTQ files (File -&gt; Open) by selecting them
both and clicking OK). Look at tabs for both files:</p>
<p><img alt="image" src="../images/paired_fastqc.png" /></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Are the quality scores the same for both files?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Overall yes</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Which value varies?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Per sequence quality scores</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Take a look at the Per base sequence quality for both files. Did you note that it is not good for either file?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>The quality score of both files drop very fast. Qualities of the REV strand drop faster than the FWD strand. This is because the template has been sat around while the FWD strand was sequenced.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>At which positions would you cut the reads if we did “fixed length trimming”?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Looking at the “Per base quality&rdquo; and “Per base sequence content”, I would choose around 27</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why does the quality deteriorate towards the end of the read?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Errors more likely for later cycles</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Does it make sense to trim the 5’ start of reads?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Looking at the “Per base sequence content&rdquo;, yes - there is a clear signal at the beginning.</p>
</details>
</div>
<p>Have a look at the other options that FastQC offers.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Which other statistics could you use to support your trimming strategy?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>“Per base sequence content&rdquo;, “Per base GC content&rdquo;, “Kmer content&rdquo;, “Per base sequence quality&rdquo;</p>
</details>
</div>
<p><img alt="image" src="../images/paired_fastqc_quality_plots.png" /></p>
<p>Once you have decided what your trim points will be, close FastQC. We
will use <code class="codehilite">fastx_trimmer</code> from the FASTX-Toolkit to perform fixed-length
trimming. For usage information see the help:</p>
<div class="codehilite"><pre><span></span>fastx_trimmer -h
</pre></div>


<p><code class="codehilite">fastx_trimmer</code> is not able to read compressed FASTQ files, so we first
need to decompress the files ready for input.</p>
<p>The suggestion (hopefully not far from your own thoughts?) is that you
trim your reads as follows:</p>
<div class="codehilite"><pre><span></span>gunzip &amp;lt; SRR023408_1.fastq.gz &amp;gt; SRR023408_1.fastq
gunzip &amp;lt; SRR023408_2.fastq.gz &amp;gt; SRR023408_2.fastq
fastx_trimmer -Q 33 -f 1 -l 32 -i SRR023408_1.fastq -o SRR023408_trim1.fastq 
fastx_trimmer -Q 33 -f 1 -l 27 -i SRR023408_2.fastq -o SRR023408_trim2.fastq
</pre></div>


<p>Many NGS read files are large. This means that simply reading and
writing files can become the bottleneck, also known as I/O bound.
Therefore, it is often good practice to avoid unnecessary disk
read/write.</p>
<div class="admonition hint">
<p class="admonition-title">Tips</p>
<p>We could do what is called pipelining to send a stream of data from one command to another, using the pipe (<code class="codehilite">|</code>) character, without the need for intermediary files. The following command would achieve this:       </p>
<p>gunzip &ndash;to-stdout &lt; SRR023408_1.fastq.gz | fastx_trimmer -Q 33 -f 4 -l 32 -o SRR023408_trim1.fastq   <br />
gunzip &ndash;to-stdout &lt; SRR023408_2.fastq.gz | fastx_trimmer -Q 33 -f 3 -l 29 -o SRR023408_trim2.fastq    </p>
</div>
<p>Now run <code class="codehilite">velveth</code> with a k-mer value of 21 for both the untrimmed and
trimmed read files in <code class="codehilite">-shortPaired</code> mode. Separate the output of the
two executions of <code class="codehilite">velveth</code> into suitably named directories, followed by
<code class="codehilite">velvetg</code>:</p>
<div class="codehilite"><pre><span></span># untrimmed reads
velveth run_21 21 -fmtAuto -create_binary -shortPaired -separate SRR023408_1.fastq SRR023408_2.fastq
time velvetg run_21

# trimmed reads
velveth run_21trim 21 -fmtAuto -create_binary -shortPaired -separate SRR023408_trim1.fastq SRR023408_trim2.fastq
time velvetg run_21trim
</pre></div>


<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How long did the two <code class="codehilite">velvetg</code> runs take?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>run_25: <code class="codehilite">real 3m16.132s; user 8m18.261s; sys 0m7.317s</code>   <br />
run_25trim: <code class="codehilite">real 1m18.611s; user 3m53.140s; sys 0m4.962s</code></p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What N50 scores did you achieve?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Untrimmed: 11   <br />
Trimmed: 15</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What were the overall effects of trimming?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Time saving, increased N50, reduced coverage</p>
</details>
</div>
<p>The evidence is that trimming improved the assembly. The thing to do
surely, is to run <code class="codehilite">velvetg</code> with the <code class="codehilite">-cov_cutoff</code> and <code class="codehilite">-exp_cov</code>. In
order to use <code class="codehilite">-cov_cutoff</code> and <code class="codehilite">-exp_cov</code> sensibly, you need to
investigate with R, as you did in the previous exercise, what parameter
values to use. Start up R and produce the weighted histograms:</p>
<div class="codehilite"><pre><span></span>    R <span class="o">--</span>no<span class="o">-</span><span class="kp">save</span>
    <span class="kn">library</span><span class="p">(</span>plotrix<span class="p">)</span> 
    data <span class="o">&lt;-</span>read.table<span class="p">(</span><span class="s">&quot;run_21/stats.txt&quot;</span><span class="p">,</span> header<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span> 
    data2 <span class="o">&lt;-</span> read.table<span class="p">(</span><span class="s">&quot;run_21trim/stats.txt&quot;</span><span class="p">,</span> header<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span> 
    par<span class="p">(</span>mfrow<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">))</span>
    weighted.hist<span class="p">(</span>data<span class="o">$</span>short1_cov<span class="p">,</span> data<span class="o">$</span>lgth<span class="p">,</span> breaks<span class="o">=</span><span class="m">0</span><span class="o">:</span><span class="m">50</span><span class="p">)</span>
    weighted.hist<span class="p">(</span>data2<span class="o">$</span>short1_cov<span class="p">,</span> data2<span class="o">$</span>lgth<span class="p">,</span> breaks<span class="o">=</span><span class="m">0</span><span class="o">:</span><span class="m">50</span><span class="p">)</span>
</pre></div>

<p>Weighted k-mer coverage histograms of the paired-end reads pre-trimmed (left) and post-trimmed
(right).</p>
<p><img alt="image" src="../images/velvet_Rplot002.png" /></p>
<p>For the untrimmed read histogram (left) there is an expected coverage of
around 13 with a coverage cut-off of around 7. For the trimmed read
histogram (right) there is an expected coverage of around 9 with a
coverage cut-off of around 5.</p>
<p>If you disagree, feel free to try different settings, but first quit R
before running <code class="codehilite">velvetg</code>:</p>
<div class="codehilite"><pre><span></span>q()

time velvetg run_21 -cov_cutoff 7 -exp_cov 13 -ins_length 92
time velvetg run_21trim -cov_cutoff 5 -exp_cov 9 -ins_length 92
</pre></div>


<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How good does it look now?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Still not great   <br />
Runtime: Reduced runtime   <br />
Memory: Lower memory usage</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>K-mer choice (Can you use k-mer 31 for a read of length 30 bp?)</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>K-mer has to be lower than the read length and the K-mer coverage should be sufficient to produce results.</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Does less data mean “worse&rdquo; results?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>Not necessarily. If you have lots of data you can safely remove poor data without too much impact on overall coverage.</p>
</details>
</div>
<p>Compare the results, produced during the last exercises, with each other, </p>
<table>
<thead>
<tr>
<th align="left">Metric</th>
<th align="right">SRR023408</th>
<th align="right">SRR023408.trimmed</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">Overall Quality (1-5)</td>
<td align="right">5</td>
<td align="right">4</td>
</tr>
<tr>
<td align="left">bp Coverage</td>
<td align="right">95x (37bp; 7761796)</td>
<td align="right">82x (32bp; 7761796)</td>
</tr>
<tr>
<td align="left">k-mer Coverage</td>
<td align="right">43x (21); 33x (25)</td>
<td align="right">30x (21); 20.5x (25)</td>
</tr>
<tr>
<td align="left">N50 (k-mer used)</td>
<td align="right">2,803 (21)</td>
<td align="right">2,914 (21)</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What would you consider as the “best&rdquo; assembly?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>SRR023408.trimmed</p>
</details>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>If you found a candidate, why do you consider it as “best&rdquo; assembly?</p>
</div>
<div class="admonition success">
<details><summary>Answer</summary></details>
<p>Overall data quality and coverage</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How else might you assess the the quality of an assembly? </p>
</div>
<div class="admonition hint">
<details><summary><strong>Hint</strong></summary><p>Hawkeye</p>
</details>
</div>
<div class="admonition success">
<details><summary>Answer</summary><p>By trying to identify paired-end constraint violations using AMOS Hawkeye.</p>
</details>
</div>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../../rna-seq/handout/handout/" title="RNA-Seq" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                RNA-Seq
              </span>
            </div>
          </a>
        
        
          <a href="../../../denovo-canu/handout/handout/" title="De Novo Canu" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                De Novo Canu
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application.5e60981f.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>